lib/SummaryUIProcessor.php

```diff
@@ -64,9 +65,14 @@ public function summaryConfig() {
             $this->summaryConfig['totalRecords'] = $this->totalRecords();
             if ($this->summaryConfig['strategy'] === ReportStrategy::ITEMIZED) {
                 $bucketBy = $this->summaryConfig['bucketBy'];
-                $bucketByLabel = $this->dataDictionary->getFieldLabel($bucketBy);
-                $this->summaryConfig['bucketByLabel'] = $bucketByLabel;
-                $this->summaryConfig['data'] = $this->mapBucketData();
```

Comment:
I'm a little concerned about the nested conditionals in this function. Can you see any way to refactor it to make it more readable?

Reply:
I agree. I felt a bit uneasy with this change. I'll give it a bit of thought. Maybe I can pull some of it out into a class. Really needs some tests.